artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPConnectionContext.java

```diff
@@ -129,7 +156,7 @@ public AMQPConnectionContext(ProtonProtocolManager protocolManager,
       } else {
          nettyExecutor = new ExecutorNettyAdapter(protocolManager.getServer().getExecutorFactory().getExecutor());
       }
-      this.handler = new ProtonHandler(nettyExecutor, protocolManager.getServer().getExecutorFactory().getExecutor(), isIncomingConnection);
+      this.handler = new ProtonHandler(nettyExecutor, protocolManager.getServer().getExecutorFactory().getExecutor(), isIncomingConnection && saslClientFactory == null);
```

Comment:
this is not actually part of my change. This was this way before since "ARTEMIS-1454: Support SASL in outgoing AMQP" on commit cc8a0cb90eba71aa595f0ed15814ae3bec326af2

Reply:
The bit im talking about is part of your change, its the diff being commented on. The change makes the value passed for ''isServer' also dependent on whether or not saslClientFactory is null or not. The usage of these bits it in the overall changes mean it can be both null and non-null values, for outgoing 'client' connections. I dont see why the value passed for 'isServer' would change depending on this.